libs/SmartSync/src/com/salesforce/androidsdk/smartsync/util/SoqlSyncDownTarget.java

```diff
@@ -130,6 +138,80 @@ public JSONArray continueFetch(SyncManager syncManager) throws IOException, JSON
         return records;
     }
 
+    @Override
+    public Set<String> getListOfRemoteIds(SyncManager syncManager, Set<String> localIds) {
```

Comment:
The implementation here is really simple now. It simply breaks down the ```SOQL``` query to get only ```Id```, and reuses ```startFetch``` and ```continueFetch``` to build records.